apps/nextra/pages/en/build/move/move-on-aptos/third-party-dependencies.mdx

```diff
@@ -0,0 +1,285 @@
+---
+title: "Third Party Dependencies"
+---
+
+import { FileTree } from "nextra/components";
+import { Callout } from 'nextra/components'
+
+
+# Third Party Dependencies
+
+Third party dependencies are external [modules](../book/modules-and-scripts.md) that a controlled module interacts with. Typically, these external modules exist under different accounts.
+
+
+## Multi-DEX router example
+
+A multi-DEX router actively utilizes third party dependencies. Instead of submitting multiple transactions to interact with different DEXs and their individual [entry](../book/functions#entry-modifier) functions within a swap route, a module (or [script](../book/modules-and-scripts.md#scripts)) can consolidate all independent DEX invocations into a single, atomic transaction. The multi-DEX router references and calls to functions present in each of the third party DEX modules to achieve this.
+
+## Sources
+
+Third party dependencies will have varying amounts of reliability and available information based on where they’re sourced from. 
+
+|  | **Git Repository** | **Local Source Code** | **Decompiler** | **Bytecode** | **ABI** | 
+| --- | --- | --- | --- | --- | --- |
+| **Requires direct support from the external module owner**  | Yes. | Yes. | No. | No. | No. | 
+| **Accuracy of dependency code** | (Possibly) verified against the on-chain bytecode. | (Possibly) verified against the on-chain bytecode. | Functionally accurate. | 1:1 bytecode. Generated interface will have correct method signatures, but logic will be missing. | Manual interface will have correct method signatures, but logic will be missing. | 
+| **Readability of dependency code** | True source code. | True source code. | Method signatures readable. Logic refactored. | Method signatures readable. No logic. | Method signatures readable. No logic. | 
+| **Aptos Prover usability** | Usable. | Usable. | Usable. | Unusable. | Unusable. | 
+| **Aptos CLI Test usability** | Usable. | Usable. |  Usable. | Unusable. | Unusable. |
+| **Necessities when third party upgrades dependent package** | Requires the repo to match on-chain bytecode. | Requires manual retrieval. | Requires the decompiler steps to be re-performed. | Requires the bytecode steps to be re-performed. | Requires the ABI steps to be re-performed. | 
```

Comment:
This may be a little confusing as a chart.

They're all reliable, but I think the important thing here is to mention that there is no documentation, and the logic may be slightly different.

Maybe we should instead say that source code allows you to do everything, but may not match on-chain state.

Onchain bytecode can be decompiled into usable source code, but without comments.

Reply:
Good points, the table did feel a bit odd upon re-reading the section... Created a second [PR](https://github.com/aptos-labs/developer-docs/pull/301) to address this.